presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/InnerJoinPredicateUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
```

Comment:
As I noted here: https://github.com/Teradata/presto/pull/555#discussion_r117824885
I wasn't sure what @sopel39 intended, so I left this as is until I get more info.